eng/pipelines/templates/jobs/archetype-sdk-tests.yml

```diff
@@ -96,7 +96,7 @@ jobs:
         condition: always()
         displayName: "Publish Results ($(TestTargetFramework))"
         inputs:
-          testResultsFiles: "**/$(TestTargetFramework)*.trx"
+          testResultsFiles: "**/$(OSName)-$(TestTargetFramework)*.trx"
```

Comment:
Are there multiple trx files any longer after this change? In a similar question do we still need the mergeTestResults: true option below? Perhaps all you are really after is to set that parameter to false?

Reply:
Yes there are separate trx files for each platform. Actually there is a trx file per project for each platform. Each trx file now has the OSName prepended. 
My understanding is that mergeTestResults makes it so that each of the trx files (1 for each test project) will be included under the same run (1 for each platform). If this was false, I think we would have multiple parent test runs per platform (1 for each test project).

Reply:
Nevermind there will still be one per test library. 

However I'm still a little unclear why it is interesting to change this name. Do we plan to download the trx files directly?

Reply:
The use case is for inspecting the output of passed tests. The output doesn't get included in the Test tabs for passed tests.

Reply:
Yeah library name would be nice to add as well.

Reply:
Is it possible to add this? I thought this run is for the whole service directory so we wouldn't have a chance to alter the name based on test project. Is there a predefined variable for this?

Reply:
Good point. Without running the dotnet test command on the individual projects we cannot easily give it a different file name per library. However I will say when I downloaded the trx zip file from the published test results they all have the same name and so you have to carefully copy each out if you want to look at them individually.

Reply:
Based on https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-vstest 

>If LogFileName isn't provided, a unique file name is created to hold the test results.

So I wonder if we should just not specify the trx file name. Sure we might get a random file name but hopefully it will be unique for each library.

Reply:
But then we wouldn't get the OS name. That said, I'm not sure of the value of the OS name given that I think you have to download the test runs one at a time. You can name the directory anything you want after downloading.
@heaths thoughts?

Reply:
You will have to click on the "OS - TestFramework" link in the uploaded test log page and then download the trx.zip file so you will know based on that. I'd suggest giving a test in your PR (or another) and see what the end result looks like.

Reply:
I requested this and would like this format because often times - to diagnose potential shared resources - I need to examine multiple logs at once. If they have the same names, it's a hassle during extraction. There are workarounds, of course, but just extracting them all to the same directory is easiest. Is there some reason we shouldn't take this change?

Reply:
Also, if you want to keep the filter simple, you could just change the file name to "*.trx" under the same directory. It's not like there'll be any others (machines are hosted and reclaimed after each run).

Reply:
The issue Wes mentioned is that the trx files will be the same for each test project. I'm not sure this is much of an issue outside of Storage.

Reply:
e.g. here is what it looks like when downloading storage test run for one platform.

![image](https://user-images.githubusercontent.com/54595583/81758945-21117500-9478-11ea-923d-37ea92b7ab71.png)

Reply:
Yeah, KV has that problem too but, as you two mentioned above (I tried to solve this and couldn't find a good solution either), there's not much we can do about that. Maybe letting the file name be random would be better. The file name isn't immediately identifiable, but it'll be in the root element (or somewhere under that toward the top - don't remember exactly off hand).

Reply:
Here is what it looks like with no filename specified. Note the prefix differs between OS, but the suffix is just the time stamp. It looks like the only thing we lose as compared to using OSName is differentiated between Core and Framework on Windows.

Linux:
![image](https://user-images.githubusercontent.com/54595583/81775233-4b2b5d00-94a1-11ea-865d-81f8f863ae88.png)

Windows Core:
![image](https://user-images.githubusercontent.com/54595583/81775322-7c0b9200-94a1-11ea-93a1-4c5e156b3832.png)

Windows Framework:
![image](https://user-images.githubusercontent.com/54595583/81775774-52069f80-94a2-11ea-96a6-860421da170a.png)

Mac:
![image](https://user-images.githubusercontent.com/54595583/81775383-9e051480-94a1-11ea-943c-d356b7edc95b.png)

Reply:
At least that would allow us to always extract the zip without conflicts and file overwrites. We will just need to ensure the zip we download is extracted into a specifically named directory based on the OS and framework.

Reply:
@pakrym @heaths are we good to merge?